subsys/net/l2/openthread/openthread.c

```diff
@@ -227,17 +239,75 @@ void ot_receive_handler(otMessage *aMessage, void *context)
 	otMessageFree(aMessage);
 }
 
+static struct openthread_context *ot_context_for_restarter;
+
+void ot_joiner_start_handler(otError error, void *context);
+
+static void ot_joiner_restarter(void *dummy)
+{
+	otError error = otJoinerStart(ot_context_for_restarter->instance,
+					joiner_pskd, NULL,
+					PACKAGE_NAME, OT_PLATFORM_INFO,
+					PACKAGE_VERSION, NULL,
+					&ot_joiner_start_handler,
+					ot_context_for_restarter);
+
+	if (error != OT_ERROR_NONE) {
+		NET_ERR("Failed to start joiner [%d]", error);
+	}
+}
+
+K_WORK_DEFINE(joiner_restart, ot_joiner_restarter);
+
+static void ot_joiner_restart_timer(void *dummy)
+{
+	k_work_submit(&joiner_restart);
```

Comment:
You could simply use `k_delayed_work_submit` instead of timer + work combo.

Reply:
We have tried that but that did not work with our zephyr devices.